### PR TITLE
test(contracts): zero-performance-fee-on-impairment regression test (#451)

### LINKED ISSUE #451: bug(contracts): missing impairment regression test — zero performance fee invariant unverified
## Summary

The vault contract correctly handles loss scenarios — when `yield_part < 0` (impairment), the performance fee is skipped. However, there is no test that proves this invariant. A future refactor of the fee calculation logic could silently break this behaviour and charge performance fees during a loss event, penalising users who are already experiencing a loss.

## Affected File

`packages/contracts/contracts/vault/src/test.rs`

## What Needs to Be Tested

The issue spec (referenced in PR #275) explicitly required this test:

> "User deposits at rate 1.0, rate halves (impairment) → zero performance fee."

The production code handles it correctly via a guard like:

```rust
if yield_part < 0 {
    // impairment — skip performance fee
    return 0;
}
```

But no test exercises this code path.

## Required Test

Add a test to `test.rs` that:

1. Initialises the vault with a known share price (e.g., `1.0`)
2. Simulates an impairment event that halves the share price (e.g., to `0.5`)
3. Calls the performance fee calculation function
4. Asserts that the performance fee returned is `0`

Example structure:

```rust
#[test]
fn test_no_performance_fee_on_impairment() {
    let env = Env::default();
    // ... setup vault with initial price = 1.0
    // ... simulate loss: price drops to 0.5
    let fee = calculate_performance_fee(&env, /* args */);
    assert_eq!(fee, 0, "performance fee must be zero when vault has experienced a loss");
}
```

## Why This Matters

- Without this test, the zero-fee invariant under impairment is only enforced by reading the code — not by CI
- Any change to `calculate_performance_fee` or related functions could break this silently
- Charging fees during a loss is a critical financial bug that directly harms users

## Acceptance Criteria

- [ ] Test added that simulates a share price drop (impairment scenario)
- [ ] Test asserts `performance_fee == 0` when `yield_part < 0`
- [ ] `cargo test -p vault-contract` passes with the new test
- [ ] Test is co-located with existing vault tests in `test.rs`

## References

- OSS_CLEANUP.md — Performance Fee Fix section (PR #275)
- The production guard exists in `packages/contracts/contracts/vault/src/lib.rs`

### PR DESCRIPTION
**Pull request by chigozirim007**
## Summary

closes #451

Adds `withdrawal_charges_no_perf_fee_on_impairment` to the vault contract tests. It deposits, simulates a 20% impairment via `report_yield(-200)`, withdraws, and asserts the user recovers the full impaired value with **no performance fee** — proving the `if yield_part > 0` guard in `lib.rs` and locking the invariant against future fee-logic refactors.

## Note on the other three issues in this batch
- **#453** (add `AUTH_JWT_SECRET` to the docker-compose API service) is **already present** in `docker-compose.yml` (line 42) — nothing to change.
- **#372** (automated vault rebalancing scheduler) and **#373** (email + push notification system) are **large feature builds** (Go services with DB models, scheduling/dispatch loops, and API/dapp wiring). I can't build/run the Go toolchain in my environment, and shipping two full features blind risks broken code — so they're intentionally **not** in this PR. They warrant a session where the repo can actually be compiled and tested.

### CODE REVIEW

File: packages/contracts/contracts/vault/src/test.rs [hunk 1]
```diff
@@ -950,3 +950,41 @@ fn process_emergency_queue_decrements_liquid_reserved() {
...
+    vault.report_yield(&admin, &(-(200 * XLM)));
+
+    let shares = vault.get_shares(&user);
+    vault.withdraw(&user, &shares, &0);
+
+    // yield_part = 800 - 1000 = -200 (< 0) => no performance fee. The user
+    // recovers the full impaired ~800 XLM, not ~780 after a wrongful 10% fee.
+    let balance = token::Client::new(&env, &token.address).balance(&user);
+    assert!(
+        balance >= 800 * XLM - 10 && balance <= 800 * XLM,
```
Comment (0xDeon) [inline packages/contracts/contracts/vault/src/test.rs]: Fuzzy bounds are unnecessary here. Soroban arithmetic is deterministic integer arithmetic — a 20% impairment on exactly 1000 XLM produces exactly 800 XLM with no rounding. Replace with: `assert_eq!(balance, 800 * XLM, "impairment must not charge a performance fee");`

**Review by 0xDeon — CHANGES_REQUESTED** [review]

File: packages/contracts/contracts/vault/src/test.rs [hunk 2]
```diff
@@ -950,3 +950,39 @@ fn process_emergency_queue_decrements_liquid_reserved() {
...
+/// Regression test for the zero-performance-fee-on-impairment invariant (#451).
+///
+/// When a vault is impaired (`yield_part < 0`), `withdraw` must NOT charge a
+/// performance fee. The fee guard is `if yield_part > 0` in `lib.rs`; a future
+/// refactor could silently drop it and start charging fees on losses. This test
+/// locks the behaviour in: with the early-withdrawal fee disabled, an impaired
+/// withdrawal returns the full impaired value with no fee skimmed off.
+#[test]
+fn withdrawal_charges_no_perf_fee_on_impairment() {
+    let (env, admin, token, vault, _treasury) = setup();
```
Comment (0xDeon) [inline packages/contracts/contracts/vault/src/test.rs:963]: This only delivers #451. The branch name (`nester-372-373-451-453`) and the PR's `closes` lines imply all four issues were scoped here, but #372 (rebalance scheduler) and #373 (notification system) are explicitly not done. Also, PR #481 is open and adds an equivalent impairment test for the same issue. Please either: expand the scope to cover the remaining issues, or scope this PR down, rename the branch, and remove the `closes` lines for the unaddressed issues so they stay open.

**Review by 0xDeon — CHANGES_REQUESTED** [review]
